teamC/dayoung/app.js

```diff
@@ -21,20 +23,40 @@ const appDataSource = new DataSource({
 }) 
 
 appDataSource.initialize()
-  .then(() => {
+    .then(() => {
       console.log("Data Source has been initialized!");
-  })
-  .catch((err) =>{
-     console.err("Error during Data Source initialization", err);
-      appDataSource.destroy()
-  });
+    })
+    .catch((err) => {
+      console.error("Error during Data Source initialization", err)
+    });
 
-app.get('/ping', function(req, res, next){
+app.get('/ping', cors(), function(req, res, next){
   res.json({message : 'pong'})
 })
 
-const server = http.createServer(app);
-const PORT = process.env.PORT;
+app.post("/users", async(req, res, next) => {
+  const { name, email, profile_image, password} = req.body;
+  await appDataSource.manager.query(
+    `SELECT *
+    FROM users
+    WHERE users.email = "${email}"`
+    ,async (err, rows) => { // async 자동으로 생성되는데 왜 그러는건가요?
+      if(Object.keys(rows).length == 0){
+        await appDataSource.query(`INSERT INTO users(
+          name, 
+          email, 
+          profile_image, 
+          password
+          )values(?, ?, ?, ?);`,
+        [name, email, profile_image, password]
+      );
+      res.status(200).json({"message" : "userCreated"});
+      }
+      else 
+      res.status(200).json({"message" : "fail"});
+    }
+  )
```

Comment:
아래와 같은 방향으로 리펙토링 해보면 좋을 것 같습니다!
그리고 말씀하신 async는 vscode에서 자동완성 해주는 것 같아요! 

함수 안에서 await를 사용하려면, 반드시 함수를 async로 감싸주어야 하기 때문이에요!

```suggestion
  const { name, email, profile_image, password} = req.body;
  
  const user = await appDataSource.query(`
    SELECT
        id,
        name,
        email,
        profile_image AS profileImage,
        password
    FROM users
    WHERE email = "${email}"`
   )
  
   if () {res.status(200).json({"message" : "fail"});}

   await appDataSource.query(`INSERT INTO users(
          name, 
          email, 
          profile_image, 
          password
          )values(?, ?, ?, ?);`,
        [name, email, profile_image, password]
      );
  res.status(200).json({"message" : "userCreated"})
    }
  )
```

Reply:
수정했습니다!